dhcp/install.sh

```diff
@@ -2,12 +2,33 @@
 
 source /vagrant/settings.sh
 
-#installation
+# installation
 apt --yes update
 apt --yes --install-recommends install dnsmasq
+export DEBIAN_FRONTEND=noninteractive
+apt -yq --yes install iptables-persistent
 
-#configuration
+# fetching interface name
+INTERNET=$(ifconfig | grep -m 1 RUNNING | cut -d':' -f1)
```

Comment:
I understand that you have copied this from some example, and in the case of example it may give a correct result, but not always.
Try this one which is safer: `ip route | grep default | cut -d' ' -f5`

Reply:
okay. I will now prefer `ip route` over `ifconfig`.

Reply:
> okay. I will now prefer ip route over ifconfig

It is not a matter of preference, your first implementation is not guaranteed to work in all the cases.